src/endpoints/sources.js

```diff
@@ -270,6 +270,56 @@ const putNarrative = contentTypesConsumed([
 ]);
 
 
+/* Group customizations
+ */
+
+/* GET
+ */
+const getGroupOverview = contentTypesProvided([
+  ["text/markdown", sendGroupOverview],
+  ["text/plain", sendGroupOverview],
+]);
+
+
+/* PUT
+ */
+const putGroupOverview = contentTypesConsumed([
+  ["text/markdown", receiveGroupOverview],
+]);
+
+/* DELETE
+ */
+const deleteGroupOverview = async (req, res) => {
+  const groupSource = req.context.source;
+
+  authz.assertAuthorized(req.user, authz.actions.Write, groupSource);
+
+  return await proxyDeleteUpstream(res,
+    async (method) => await groupSource.urlFor("group-overview.md", method)
+  );
+};
+
+/**
+ * Send a DELETE request to the upstream URL.
+ *
+ * @param {express.response} res - Express-style response instance
+ * @param {upstreamUrlExtractor} upstreamUrlExtractor - Callback to retrieve resource URL
+ */
+// TODO: use this function in deleteResource to avoid code duplication?
+async function proxyDeleteUpstream(res, upstreamUrlExtractor) {
```

Comment:
Replying to https://github.com/nextstrain/nextstrain.org/pull/570#issuecomment-1225010406 in a "thread".

> Organizationally, I'm not sure it makes sense to have `proxyDeleteUpstream` as-is. I'd probably either a) inline it into `deleteGroupOverview` (like the similar inline in `deleteResource`) or b) refactor `deleteResource`'s multi-URL implementation into a `delete(urls: String[])` function and use it in both `deleteResource` and `deleteGroupOverview`. Happy to sketch out either or both tomorrow if that'd be clearer.

That makes sense. I did a sketch of both: (a) f5acfc7c4b9e1b3aa75a5192dff2521b2932da52 (b) 13ff607117fef230cf0eb9eb60cdd3418beb16d7

Here's some pros and cons to both as-is:

- f5acfc7c4b9e1b3aa75a5192dff2521b2932da52
    - Pro: has a custom error message specifying what failed to be deleted.
    - Con: duplicate logic around `fetch` and checking the response status.
- 13ff607117fef230cf0eb9eb60cdd3418beb16d7
    - Pro: less code duplication.
    - Con: error message is generic: `At least one delete request failed.`

Maybe less code duplication is preferred, otherwise it would be duplicated 3 times (`deleteResource`, `deleteGroupOverview`, `deleteGroupLogo`) and potentially more in the future.

To mitigate the con of generic error messages, we could add a parameter `deleteByUrls(urls, errorMessage)` but not sure if that's too extra.

Reply:
I like the refactored version in 13ff607. I don't think we need a second `errorMessage` parameter. Instead, we could extend the `deleteByUrls` function to produce a more specific error message with the URL(s) that failed and their statuses. Also note that, IIRC, the error message will be for our eyes only during troubleshooting/debugging; it won't be shown to the end user, who will just see a generic "server error" message of some sort. So any error message improvements we make here are for our own sake, and we could punt on the errors now at the expensive of our future selves. ;-)

Reply:
Ok, I'll punt on it for now.

Maybe something like 34f02d57e29d99d2725fa7bb2dc20a2a8ee49866 can work if we want failed URLs and statuses.